test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5139,6 +5132,87 @@ def revert_to_defaults(self, reverthooks=True, revertqueues=True,
                 self.manager(MGR_CMD_DELETE, RSC, id=rescs)
         return True
 
+    def unset_svr_attrib(self, server_stat=None):
+        """
+        Unset server attributes
+        """
+        ignore_attrs = ['id', 'pbs_license', ATTR_NODE_ProvisionEnable]
+        ignore_attrs += [ATTR_status, ATTR_total, ATTR_count]
+        ignore_attrs += [ATTR_rescassn, ATTR_FLicenses, ATTR_SvrHost]
+        ignore_attrs += [ATTR_license_count, ATTR_version, ATTR_managers]
+        ignore_attrs += [ATTR_pbs_license_info, ATTR_power_provisioning]
+        unsetlist = []
+        self.cleanup_jobs_and_reservations()
+        if server_stat is None:
+            server_stat = self.status(SERVER, level=logging.DEBUG)[0]
+        for k in server_stat.keys():
+            if (k in ignore_attrs) or (k in self.dflt_attributes.keys()):
+                continue
+            elif (('.' in k) and (k.split('.')[0] in ignore_attrs)):
+                continue
+            else:
+                unsetlist.append(k)
+        if len(unsetlist) != 0:
+            self.manager(MGR_CMD_UNSET, MGR_OBJ_SERVER, unsetlist)
+
+    def delete_site_hooks(self):
+        """
+        Delete site hooks from PBS
+        """
+        skip_site_hooks = ['pbs_cgroups']
+        hooks = self.status(HOOK, level=logging.DEBUG)
+        hooks = [h['id'] for h in hooks]
+        for h in skip_site_hooks:
+            if h in hooks:
+                hooks.remove(h)
+        if len(hooks) > 0:
+            self.manager(MGR_CMD_DELETE, HOOK, id=hooks)
+
+    def delete_queues(self):
+        """
+        Delete queues
+        """
+        queues = self.status(QUEUE, level=logging.DEBUG)
+        queues = [q['id'] for q in queues]
+        if len(queues) > 0:
+            try:
+                nodes = self.status(VNODE, logerr=False)
+                for node in nodes:
+                    if 'queue' in node.keys():
+                        self.manager(MGR_CMD_UNSET, NODE, 'queue',
+                                     node['id'])
+            except BaseException:
+                pass
+            self.manager(MGR_CMD_DELETE, QUEUE, id=queues)
+
+    def delete_sched_config(self):
+        """
+        Delete sched_priv & sched_log files
+        """
+        self.manager(MGR_CMD_LIST, SCHED)
+        for name in self.schedulers.keys():
```

Comment:
Please do list(self.schedulers.keys()) here as you are modifying the object at line 5195.

Reply:
thanks